python/cudf_polars/cudf_polars/experimental/rapidsmpf/core.py

```diff
@@ -86,27 +87,48 @@ def evaluate_logical_plan(
     assert config_options.executor.name == "streaming", "Executor must be streaming"
     assert config_options.executor.runtime == "rapidsmpf", "Runtime must be rapidsmpf"
 
-    if (
-        config_options.executor.scheduler == "distributed"
-    ):  # pragma: no cover; Requires distributed
-        # TODO: Add distributed-execution support
-        raise NotImplementedError(
-            "The rapidsmpf engine does not support distributed execution yet."
-        )
-
     # Lower the IR graph on the client process (for now).
     ir, partition_info, stats = lower_ir_graph(ir, config_options)
 
-    # Reserve collective IDs for the entire pipeline execution
-    with ReserveOpIDs(ir) as collective_id_map:
-        result, metadata_collector = evaluate_pipeline(
-            ir,
-            partition_info,
-            config_options,
-            stats,
-            collective_id_map,
-            collect_metadata=collect_metadata,
-        )
+    # Reserve shuffle IDs for the entire pipeline execution
+    with ReserveOpIDs(ir) as shuffle_id_map:
+        # Build and execute the streaming pipeline.
+        # This must be done on all worker processes
+        # for cluster == "distributed".
+        if (
+            config_options.executor.cluster == "distributed"
+        ):  # pragma: no cover; block depends on executor type and Distributed cluster
+            # Distributed execution: Use client.run
+
+            # Warn loudly that multi-GPU execution is under construction
+            warnings.warn(
```

Comment:
IMO, we should have a way to run this without leaking a warning.

And I'd prefer it if all the statically know configuration issues / warnings could be caught when constructing the `ConfigOptions` rather than further down here.